### PR TITLE
Docker: support new NVidia runtime

### PR DESCRIPTION
**Pull request by AdamGleave**
## Description

The newest version of [NVidia Docker](https://github.com/NVIDIA/nvidia-docker/wiki/Installation-(Native-GPU-Support)) has a new syntax. However, older versions are still in widespread use and are likely to be for the next year or two. I have updated the script to auto-detect which syntax is needed.

## Motivation and Context

- [X] I have raised an issue to propose this change ([required](https://github.com/hill-a/stable-baselines/blob/master/CONTRIBUTING.md) for new features and bug fixes)
Addresses @rracinskij comment in issue #632.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [X] I've read the [CONTRIBUTION](https://github.com/hill-a/stable-baselines/blob/master/CONTRIBUTING.md) guide (**required**)
- [X] I have updated the [changelog](https://github.com/hill-a/stable-baselines/blob/master/docs/misc/changelog.rst) accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [X] I have ensured `pytest` and `pytype` both pass.

### CODE REVIEW

File: scripts/run_docker_gpu.sh [hunk 1]
```diff
@@ -6,7 +6,14 @@ cmd_line="$@"
 echo "Executing in the docker (gpu image):"
 echo $cmd_line
 
+# TODO: always use new-style once sufficiently widely used (probably 2021 onwards)
+if [ -x "$(which nvidia-docker)" ]; then
```
Comment (araffin) [inline scripts/run_docker_gpu.sh]: so here, you are just checking if `nvidia-docker` exists?

Comment (Miffyli) [reply scripts/run_docker_gpu.sh]: Yes, or more accurately "if `nvidia-docker` is executable"

**Comment by ChengYen-Tang** [comment]
As a reminder, with the new NVidia runtime, the docker build cannot use the parameter --gpus all. 

Because the stable-baselines setup.py will test the GPU version of tensorflow, so there will be some problems in building the GPU version image of stable-baselines.

**Comment by AdamGleave** [comment]
> As a reminder, with the new NVidia runtime, the docker build cannot use the parameter `--gpus all`.
> 
> Because the stable-baselines setup.py will test the GPU version of tensorflow, so there will be some problems in building the GPU version image of stable-baselines.

I think the build should still work since the `Dockerfile` explicitly installs `tensorflow` or `tensorflow-gpu` before installing `stable_baselines` via `pip install .`

**Comment by AdamGleave** [comment]
Well, this ended up being more complicated than I expected.

@ChengYen-Tang is right: without NVidia runtime, the `import tensorflow` in `setup.py` fails when we've preinstalled `tensorflow-gpu`, and then it tries to install the CPU version of TensorFlow. I've fixed this by making `setup.py` respect the `USE_GPU` environment flag. One side-effect of this is the latest build scripts will use the latest version of TensorFlow: I dunno if there was a reason we were using `1.8.0` in Docker.

I think there was a separate bug caused by inconsistent use of `/root/code/stable-baselines` and /root/code/stable_baselines` which is what was causing the `pip install -e .` to fail. I can now mount new code in and it seems to be importable from any directory:

```
$ ./scripts/run_docker_gpu.sh /bin/bash                                                                                                  
Executing in the docker (gpu image):                                                                                                                                         
/bin/bash
# cd /
# python                                                                                                                                                      
Python 3.5.2 (default, Oct  8 2019, 13:06:37)                                                                                                                                
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import stable_baselines                                                                                                                                                  
WARNING:tensorflow:                                                                                                                                                          
The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
  * https://github.com/tensorflow/io (for I/O related ops)
If you depend on functionality not listed there, please file an issue.

>>> stable_baselines.PPO2                                                                                                                                                    
<class 'stable_baselines.ppo2.ppo2.PPO2'> 
```

@ChengYen-Tang can you confirm if the Docker images with the latest build scripts work for you? If so we can maybe close #654

**Comment by ChengYen-Tang** [comment]
@AdamGleave looks no problem, I closed it #654

File: scripts/run_docker_gpu.sh (hunk 1 above)
Comment (araffin) [reply scripts/run_docker_gpu.sh]: so if I get it right, in the newest docker version, there is no need for an additional executable?

File: Dockerfile [hunk 2]
```diff
@@ -25,22 +25,15 @@ RUN apt-get -y update \
...
-    cd $CODE_DIR && \
     pip install --upgrade pip && \
-    if [ "$USE_GPU" = "True" ]; then \
-        TENSORFLOW_PACKAGE="tensorflow-gpu==1.8.0"; \
-    else \
-        TENSORFLOW_PACKAGE="tensorflow==1.8.0"; \
-    fi; \
-    pip install ${TENSORFLOW_PACKAGE} && \
+    cd ${CODE_DIR}/stable-baselines && \
+    python setup.py egg_info && cat stable_baselines.egg-info/requires.txt && rm -rf stable_baselines.egg-info && \
```
Comment (araffin) [inline Dockerfile]: This is to check that the right tensorflow version will be installed?

Comment (AdamGleave) [reply Dockerfile]: this was debugging code I added, removed now

**Pull request merged**